test/compress/unicode.js

```diff
@@ -25,7 +26,19 @@ unicode_escaped_identifier: {
     expect_exact: 'var a="foo";var \u{10000}="bar";';
 }
 
+unicode_escaped_identifier_es5: {
+    beautify = {ecma: 5}
+    input: `
+        var \u{10000} = "bar";
```

Comment:
In ES5 mode, identifiers aren't parsed as whole code points. Instead, each individual UTF16 char valid as an identifier. Eg, `\u{10000}` is actually interpreted as two chars `\ud800` and `\udc00`. Both of those must be individually valid, not the combined code point.